src/server/scripts/Kalimdor/zone_mulgore.cpp

```diff
@@ -0,0 +1,139 @@
+#include "ScriptedCreature.h"
+#include "ScriptedGossip.h"
+#include "Player.h"
+#include "SpellInfo.h"
+
+enum eAgitatedEarthSpirit
+{
+    SPELL_SOOTHE_EARTH_SPIRIT       = 69453,
+    SPELL_ROCK_BARRAGE              = 81305,
+
+    NPC_EARTH_SPIRIT_CREDIT_BUNNY   = 36872,
+
+    EVENT_ROCK_BARRAGE              = 1
+};
+
+// 36845 - Agitated Earth Spirit
+class npc_agitated_earth_spirit : public CreatureScript
+{
+public:
+    npc_agitated_earth_spirit() : CreatureScript("npc_agitated_earth_spirit") { }
+
+    struct npc_agitated_earth_spiritAI : public ScriptedAI
+    {
+        npc_agitated_earth_spiritAI(Creature* creature) : ScriptedAI(creature) { }
+
+        void Reset() OVERRIDE
+        {
+            events.Reset();
+        }
+
+        void SpellHit(Unit* caster, SpellInfo const* spell) OVERRIDE
+        {
+            if (spell->Id == SPELL_SOOTHE_EARTH_SPIRIT)
+            {
+                Position pos;
+                caster->GetNearPoint(caster, pos.m_positionX, pos.m_positionY, pos.m_positionZ, 0.0f, MIN_MELEE_REACH, caster->GetAngle(me));
+                me->GetMotionMaster()->MovePoint(1, pos);
+                _playerGUID = caster->GetGUID();
+            }
+        }
+
+        void MovementInform(uint32 type, uint32 pointId) OVERRIDE
+        {
+            if (type == POINT_MOTION_TYPE && pointId == 1)
+            {
+                switch (urand(0, 1))
+                {
+                    case 0:
+                    {
+                        me->setFaction(35);
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                            player->KilledMonsterCredit(NPC_EARTH_SPIRIT_CREDIT_BUNNY);
+
+                        me->DespawnOrUnsummon(1000);
+                        break;
+                    }
+                    case 1:
+                        me->setFaction(14);
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                            AttackStart(player);
+                        break;
+                }
+            }
+        }
+
+        void EnterCombat(Unit* /*victim*/) OVERRIDE
+        {
+            events.ScheduleEvent(EVENT_ROCK_BARRAGE, urand(4000, 5000));
+        }
+
+        void UpdateAI(uint32 diff) OVERRIDE
+        {
+            if (!UpdateVictim())
+                return;
+
+            events.Update(diff);
+
+            while (uint32 eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_ROCK_BARRAGE:
+                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
+                            DoCast(target, SPELL_ROCK_BARRAGE);
+                        events.ScheduleEvent(EVENT_ROCK_BARRAGE, urand(18000, 21000));
+                        break;
+                }
+            }
+
+            DoMeleeAttackIfReady();
+        }
+
+    private:
+        uint64 _playerGUID;
```

Comment:
missing initialization?

Reply:
> 
> 
> missing initialization?

Line 70

Reply:
on line 70 it is set but you need to initialize it at the declaration as 0